google/cloud/sql/integration_tests/sql_integration_test.cc

```diff
@@ -97,21 +99,30 @@ TEST_F(SqlIntegrationTest, CreateEphemeral) {
   ASSERT_NE(instances.begin(), instances.end());
   ASSERT_STATUS_OK(*instances.begin());
 
-  google::cloud::sql::v1::SqlInstancesCreateEphemeralCertRequest request;
-  request.set_project(project_id_);
-  request.set_instance((*instances.begin())->name());
-  request.mutable_body()->set_public_key("THE_PUBLIC_KEY");
+  // All the RPCs that could repro #12112 are non-idempotent. So we put a little
+  // manual retry loop around the test.
+  auto const attempt_delay = std::chrono::seconds(30);
+  for (int attempt = 0; attempt != 3; ++attempt) {
+    if (attempt != 0) std::this_thread::sleep_for(attempt_delay);
+    google::cloud::sql::v1::SqlInstancesCreateEphemeralCertRequest request;
+    request.set_project(project_id_);
+    request.set_instance((*instances.begin())->name());
+    request.mutable_body()->set_public_key("THE_PUBLIC_KEY");
 
-  auto result = client.CreateEphemeral(request);
-  EXPECT_THAT(
-      result,
-      AnyOf(
-          StatusIs(StatusCode::kInvalidArgument,
-                   HasSubstr("Provided public key was in an invalid or "
-                             "unsupported format")),
-          StatusIs(
-              StatusCode::kPermissionDenied,
-              HasSubstr("The client is not authorized to make this request"))));
+    auto result = client.CreateEphemeral(request);
+    // This is a transient error, just retry.
+    if (result.status().code() == StatusCode::kUnavailable) continue;
```

Comment:
Should we continue to make the expectation on the final attempt?

Reply:
Good point.  I tried to clean this up, let me know what you think.

Reply:
Looks good.  Opt:  My only parting suggestion would be to maybe return the final `kUnavailable` status, rather than consing up a new `kDeadlineExceeded` one.